posts/2023-06-09-setting-up-a-home-energy-monitoring-system.adoc

```diff
@@ -0,0 +1,711 @@
+
+image::23-08.PNG[5 minute averages over one day on August 23rd]
+
+The next one shows the same graphs over a period of 16 days (from August 8th till August 23rd):
+
+image::08-16days.PNG[Production and consumption data over 16 consecutive days]
+
+The graph shows the granularity changes from 5 minutes to an hour on August 8th.
+
+The obvious question I asked myself was how to keep granular data longer, and I saw two options. First, I could ask the company behind the system to reinstate the longer data retention period. Second, I could investigate if I could build an energy monitoring system that would allow me to configure the data retention policy as I wished it to be. I chose option two, as it seemed to give me an opportunity to learn technologies I had never used before.
+
+Before I found out about the data granularity issue, I had already discovered that the _Smappee_ energy management system was quite open. It provides two ways to get access to the collected data. One method is to use the _Smappee_ REST API, the other is to use https://en.wikipedia.org/wiki/MQTT[_MQTT_].
+
+The REST API can query endpoints in the _Smappee_ Cloud, linked to an individual _Smappee_ monitoring system located, for example, in one’s house. Use of the API is free for non-commercial use and requires setting up https://oauth.net/2[OAuth2] authentication.
+
+When using _MQTT_, the _Smappee_ monitoring system configuration has to be tweaked to point to a so-called _MQTT Broker_. Once that’s done, the _Smappee_ device will send measurement samples at a once per second interval to the _MQTT Broker_.
+
+The first access method polls data from the _Smappee_ Cloud, while the second pushes data to a server.
+
+With all this in mind, let’s consider the project goals I set for my Home Energy Monitoring System (_HEMS_).
+
+[id=project-objectives]
+== Project objectives
+
+The primary aim of the _HEMS_ project was to collect _Smappee_ data at specific aggregation levels and retention periods. Also, we want dashboards that display selected aggregates over a chosen time period.
+
+For example, a _Smappee_, when configured to send data to an _MQTT Broker_ sends data at a one-second interval. One may want to aggregate this data at minutes, days, weeks, or months intervals. Each aggregation level may have specific data retention periods. For example, we may keep the high frequency 1 minute aggregates for a few months. On the other side of the spectrum, we may keep indefinitely the monthly averages (and obviously, as long as storage capacity permits).
+
+As I wanted to be independent of external factors (such as the _Smappee_ Cloud) as to not be exposed to the whims of external parties, I chose the _MQTT_ route for building my _HEMS_.
+
+
+Next to the already cited main objective, there were other objectives I wanted to achieve:
+
+* If possible, implement the system without any programming.
+* Use _Open Source Software_ (OSS) only.
```

Comment:
OSS in the IT system management and monitoring world usually means `Operational Support System`. Your HEMS can be considered as an Operational Support System. I think using `FOSS (Free Open Source Software)` here would make this distinction clear.

Reply:
Changed `OSS` to `FOSS` everywhere in the doc.